src/main/java/games/strategy/util/Version.java

```diff
@@ -167,29 +174,45 @@ private int compareTo(final Version other, final boolean ignoreMicro) {
     return 0;
   }
 
+  /**
+   * Returns true, if the provided Version object is greater than this object.
+   */
   public boolean isGreaterThan(final Version other) {
     return isGreaterThan(other, false);
   }
 
+  /**
+   * Returns true, if the provided Version object is greater than this object.
+   * If ignoreMicro is set to true, the micro version number is ignored.
+   */
   public boolean isGreaterThan(final Version other, final boolean ignoreMicro) {
     return compareTo(other, ignoreMicro) < 0;
   }
 
+  /**
+   * Returns true, if the provided Version object is less than this object.
+   */
   public boolean isLessThan(final Version other) {
-    return compareTo(other, false) > 0;
-  }
-
-  public String toStringFull(final String separator) {
-    return toStringFull(separator, false);
+    return compareTo(other) > 0;
   }
 
-  private String toStringFull(final String separator, final boolean noMicro) {
-    return m_major + separator + m_minor + separator + m_point
-      + (noMicro ? "" : (separator + (m_micro == Integer.MAX_VALUE ? "dev" : m_micro)));
+  /**
+   * Creates a complete version string, even if some version numbers are 0.
+   */
+  public String toStringFull() {
+    return Joiner.on('.').join(m_major, m_minor, m_point, m_micro == Integer.MAX_VALUE ? "dev" : m_micro);
```

Comment:
Will `java.lang.String#join()` or `java.util.StringJoiner` not work here?  Just trying to avoid the Guava dependency if the functionality is available in the JDK.

Reply:
In theory yes, but this would require us to put a `String.valueOf` around every var

Reply:
Ah, yes.  I didn't bother to look at the signatures to see that they both require a `CharSequence`.